### PR TITLE
dash/edit-mode-markup

### PR DESCRIPTION
**Pull request by sebastianbochan**
Removed unnecessary markup when editMode is disabled

### CODE REVIEW

File: ts/Dashboards/Board.ts [hunk 1]
```diff
@@ -265,6 +246,12 @@ class Board implements Serializable<Board, Board.JSON> {
      * */
     public guiEnabled: (boolean|undefined);
 
+    /**
+     * Flag to determine if the GUI is enabled.
+     * @internal
+     * */
+    public editModeEnabled: (boolean|undefined);
```
Comment (karolkolodziej) [inline ts/Dashboards/Board.ts:253]: use `?` instead of undefined 
also, I guess it can be private?

File: ts/Dashboards/Board.ts [hunk 2]
```diff
@@ -831,6 +851,9 @@ namespace Board {
             },
             layouts: []
         },
+        editMode: {
+            enabled: true
```
Comment (karolkolodziej) [inline ts/Dashboards/Board.ts:855]: Change the API default (EditMode.ts file) and if possible move this piece of code there.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Please add tests

File: ts/Dashboards/Board.ts (hunk 2 above)
Comment (sebastianbochan) [reply ts/Dashboards/Board.ts]: I think that it's unnecessary, so we can remove this value and leave like it was.

File: samples/dashboards/cypress/edit-mode-disabled-markup/demo.mjs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    components: [{
+        renderTo: 'dashboard-col-1',
+        type: 'HTML',
+        elements: [{
+            tagName: 'h1',
+            textContent: 'Placeholder text'
+        }]
+    }]
+});
```
Comment (Vazonik) [inline samples/dashboards/cypress/edit-mode-disabled-markup/demo.mjs:23]: missed trailing line

```suggestion
});

```

File: test/cypress/dashboards/integration/EditMode/markup.cy.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  });
+
+  it('Disabled EditMode markup should not be printed', () => {
+    cy.get('.highcharts-dashboards--fullscreen').should('not.exist');
+    cy.get('.highcharts-dashboards-edit-tools').should('not.exist');
+    cy.get('.highcharts-dashboards-edit-ctx-detection-pointer').should('not.exist');
+    cy.get('.highcharts-dashboards-edit-confirmation-popup').should('not.exist');
+    cy.get('.highcharts-dashboards-edit-overlay').should('not.exist');
+  });
+});
```
Comment (Vazonik) [inline test/cypress/dashboards/integration/EditMode/markup.cy.js:14]:
```suggestion
});

```

**Review by Vazonik — APPROVED** [review]
Looks good!

**Review by jomar-honsi — APPROVED** [review]

**Pull request merged**